File: server/controllers/definitions_controller.rb
#Controller responsável por enviar as definições do jogo aos jogadores conectados
class DefinitionsController < AppController
  
  def initialize(app)
    super(app)
    @defs = Definitions.get_instance #pega singleton Definitions
  end  
  
  #define nome do controller
  def name
    :definitions
  end
  
  #envia definições do jogo quando um novo jogador se conecta
  def new_conn(conn)
    get_colors(conn)
    get_territories(conn)
    get_regions(conn)
  end
  
  #envia lista de cores
  def get_colors(conn)
    conn.send_msg(Message.new('definitions','set_colors',@defs.colors))
  end
  
  #envia lista de territorios
  def get_territories(conn)
    conn.send_msg(Message.new('definitions','set_territories',@defs.territories))
  end
  
  #envia lista de regioes
  def get_regions(conn)
    conn.send_msg(Message.new('definitions','set_regions',@defs.regions))
  end
  
end

File: client/js/controllers/definitionscontroller.js
/* Controlador que recebe as definições do jogo, como territorios, cartas, cores... */

  var DefinitionsController;

  DefinitionsController = (function() {

    function DefinitionsController(app) {
      this.app = app;
      this.controllerName = 'definitions';
      this.colors = null;
      this.territories = null;
      this.regions = null;
    }

    DefinitionsController.prototype.set_colors = function(msg) {
      return this.colors = msg.colors;
    };

    DefinitionsController.prototype.set_territories = function(msg) {
      return this.territories = msg.territories;
    };

    DefinitionsController.prototype.set_regions = function(msg) {
      return this.regions = msg.regions;
    };

    DefinitionsController.prototype.get_colors = function() {
      return this.colors;
    };

    DefinitionsController.prototype.get_territories = function() {
      return this.territories;
    };
    
    DefinitionsController.prototype.get_regions = function() {
      return this.regions;
    };
    

    return DefinitionsController;

  })();


